lib/libzfs/libzfs_pool.c

```diff
@@ -2636,13 +2634,19 @@ zpool_vdev_is_interior(const char *name)
 }
 
 nvlist_t *
-zpool_find_vdev(zpool_handle_t *zhp, const char *path, boolean_t *avail_spare,
+zpool_find_vdev(zpool_handle_t *zhp, const char *ipath, boolean_t *avail_spare,
     boolean_t *l2cache, boolean_t *log)
 {
 	char *end;
+	const char *path;
 	nvlist_t *nvroot, *search, *ret;
 	uint64_t guid;
+	int firstpass;
+	char buf[MAXPATHLEN+16];
```

Comment:
We ran in to the same issue with short names on Linux but handled it a little bit differently.  Instead of retrying here we ended up updating `vdev_to_nvlist_iter()` code to be smarter about matching short path names.  Can you use this same logic, see `zfs_strcmp_pathname()->zfs_strcmp_shortname()`.  #9356 moves some of these functions to platform specific code so they easily do something FreeBSD specific.  Though it would be nice if they could be generic.

```
                /*
                 * Search for the requested value. Special cases:
                 *
                 * - ZPOOL_CONFIG_PATH for whole disk entries.  These end in
                 *   "-part1", or "p1".  The suffix is hidden from the user,
                 *   but included in the string, so this matches around it.
                 * - ZPOOL_CONFIG_PATH for short names zfs_strcmp_shortname()
                 *   is used to check all possible expanded paths.
                 * - looking for a top-level vdev name (i.e. ZPOOL_CONFIG_TYPE).
                 *
                 * Otherwise, all other searches are simple string compares.
                 */
                if (strcmp(srchkey, ZPOOL_CONFIG_PATH) == 0) {
                        uint64_t wholedisk = 0;
        
                        (void) nvlist_lookup_uint64(nv, ZPOOL_CONFIG_WHOLE_DISK,
                            &wholedisk);
                        if (zfs_strcmp_pathname(srchval, val, wholedisk) == 0)
                                return (nv);
```

Reply:
It looks like that needs to be done in libzutil so I'll just revert the change here.